### DISCUSSION TITLE
Add Support for extracting all images from a HEIF container

### DISCUSSION THREAD

**Discussion by 06needhamt** [ImageMagick/ImageMagick#1938] [General] [answered]
Libheif has the ability to extract all of the images out of a HEIF container, however, ImageMagick only extracts the primary image. Would it be possible to create a coder which returns an array of pointers to all the top-level images in a HEIF container?

I would be interested in trying to implement this myself, but I would need some pointers on where to look as I am not familiar with the codebase.

**Comment by dlemstra** [comment, marked as answer]
Which version are you using? The most recent version of ImageMagick 7 has support for this.

**Reply by 06needhamt** [reply]
I was using version 7.0.10-7 beta via the Magick.Net wrapper. Does the .NET version have support for this? If so how can it be achieved and if not I will open a discussion on that repository?

Thanks for your quick response.

**Reply by fmw42** [reply]
Also what version of libheif?

**Reply by dlemstra** [reply]
Are you using a `MagickImageCollection` @06needhamt? Feel free to open an issue/discussion with a code sample in the Magick.NET repository if you need more help.

**Reply by 06needhamt** [reply]
No I'm not currently. I'll update my code to do so and will report back if I have any further questions.

**Reply by 06needhamt** [reply]
Managed to get it working. Thanks for your help
